### ISSUE TITLE
WT should inject GUIDs for guidless custom profiles

### ISSUE THREAD

**Issue by dev-nicolaos** [microsoft/terminal#2876]
# Environment

```none
Windows build number: 10.0.18362.0
Windows Terminal version: 0.5.2661.0
```

# Steps to reproduce
1. Open WT
1. Open profiles.json
1. Create an object in the `profiles` array
1. Fill in some fields such as `name`, `commandline`, `startingDirectory`, `colorScheme`, etc.
1. Save profiles.json

# Expected behavior
WT will generate/add a guid to the new custom profile, that way it can be set as the `defaultProfile`. This was the case in previous versions of WT, before separation of cascading settings was implemented.

# Actual behavior
WT does not generate/add a guid to the new profile, thus there appears to be no way to set your custom profile as the default.

**Comment by DHowett-MSFT** [comment]
Curious. Thanks for teh catch! For now, if you generate a guid and set it for that guid-less profile you can set _that_ as a default.
Sorry for the miss.

**Comment by DHowett-MSFT** [comment]
This probably came about in 0.4 when we stopped writing folks' settings files back out to disk, honestly.

**Comment by DHowett-MSFT** [comment]
I've converted this into a task for injecting the GUID into the guidless profile. Thanks for reporting!

**Comment by DHowett-MSFT** [comment]
Continuing my inability to decide about this bug, I think it's better if we simply move away from GUIDs for user-facing fixtures as much as _possible_. They're great internal identifier. They're awful external ones.

I'm going to mark this, the incidental issue that this issue actually implies ("i can't set default a guid-less profile"), as fixed by #5706.
